src/mem3/src/mem3_sync_event_listener.erl

```diff
@@ -293,17 +295,38 @@ should_terminate(Pid) ->
         ok
     end).
 
-capture(Pid) ->
+
+get_state(Pid) ->
     Ref = make_ref(),
+    Pid ! {get_state, Ref, self()},
+    receive
+        {Ref, State} -> State
+    after 10 ->
+        timeout
+    end.
+
+
+wait_state_frequency(Pid, Val) ->
     WaitFun = fun() ->
-        Pid ! {get_state, Ref, self()},
-        receive
-            {Ref, State} -> State
-        after 0 ->
-            wait
+        case get_state(Pid) of
+            timeout ->
+                wait;
+            #state{frequency = Val} ->
+                true
         end
     end,
     test_util:wait(WaitFun).
 
 
+wait_state_delay(Pid, Val) ->
+    WaitFun = fun() ->
+        case get_state(Pid) of
+            timeout ->
+                wait;
+            #state{delay = Val} ->
+                true
+        end
+    end,
+    test_util:wait(WaitFun).
```

Comment:
Just a nit, but there's a lot of repetition between `wait_state_frequency` and `wait_state_delay`. You can't just parameterize the field name since it's a compile time construct, so I'm not sure there's a trivial way to DRY this up (and FTR I don't consider switching to maps trivial).

Reply:
Could probably do something there by passing #state.delay and #state.frequency where it but I think matching and getting the state part would be uglier a bit.